Sources/NIOHTTP2/ConnectionStateMachine/ConnectionStateMachine.swift

```diff
@@ -0,0 +1,1307 @@
+        case idle(IdleConnectionState)
+
+        /// Our preface has been sent, and we are awaiting the preface from the remote peer. In general we're more
+        /// likely to enter this state as a client than a server, but users may choose to reduce latency by
+        /// aggressively emitting the server preface before the client preface has been received. In either case,
+        /// in this state we are waiting for the remote peer to send its preface.
+        case prefaceSent(PrefaceSentState)
+
+        /// We have received a preface from the remote peer, and we are waiting to send our own preface. In general
+        /// we're more likely to enter this state as a server than as a client, but remote peers may be attempting
+        /// to reduce latency by aggressively emitting the server preface before they have received our preface.
+        /// In either case, in this state we are waiting for the local user to emit the preface.
+        case prefaceReceived(PrefaceReceivedState)
+
+        /// Both peers have exchanged their preface and the connection is fully active. In this state new streams
+        /// may be created, potentially by either peer, and the connection is fully useable.
+        case active(ActiveConnectionState)
+
+        /// The remote peer has sent a GOAWAY frame that quiesces the connection, preventing the creation of new
+        /// streams. However, there are still active streams that have been allowed to complete, so the connection
+        /// is not entirely inactive.
+        case remotelyQuiesced(RemotelyQuiescedState)
+
+        /// The local user has sent a GOAWAY frame that quiesces the connection, preventing the creation of new
+        /// streams. However, there are still active streams that have been allowed to complete, so the connection
+        /// is not entirely inactive.
+        case locallyQuiesced(LocallyQuiescedState)
+
+        /// Both peers have emitted a GOAWAY frame that quiesces the connection, preventing the creation of new
+        /// streams. However, there are still active streams that have been allowed to complete, so the connection
+        /// is not entirely inactive.
+        case bothQuiescing(BothQuiescingState)
+
+        /// We have sent our preface, and sent a GOAWAY, but we haven't received the remote preface yet.
+        /// This is a weird state, unlikely to be encountered in most programs, but it's technically possible.
+        case quiescingPrefaceSent(QuiescingPrefaceSentState)
+
+        /// We have received a preface, and received a GOAWAY, but we haven't sent our preface yet.
+        /// This is a weird state, unlikely to be encountered in most programs, but it's technically possible.
+        case quiescingPrefaceReceived(QuiescingPrefaceReceivedState)
+
+        /// The connection has completed, either cleanly or with an error. In this state, no further activity may
+        /// occur on the connection.
+        case closed
```

Comment:
I think we should rename this because the connection isn't actually closed. Maybe `fullyQuiesced` or `becameUnusable` or something